public/app/plugins/panel/histogram/Histogram.tsx

```diff
@@ -132,22 +154,24 @@ const prepConfig = (frame: DataFrame, theme: GrafanaTheme2) => {
     scaleKey: 'x',
     isTime: false,
     placement: AxisPlacement.Bottom,
-    incrs: histogramBucketSizes,
-    splits: xSplits,
-    values: (u: uPlot, splits: any[]) => {
-      const tickLabels = splits.map(xAxisFormatter);
-
-      const maxWidth = tickLabels.reduce(
-        (curMax, label) => Math.max(measureText(label, UPLOT_AXIS_FONT_SIZE).width, curMax),
-        0
-      );
-
-      const labelSpacing = 10;
-      const maxCount = u.bbox.width / ((maxWidth + labelSpacing) * devicePixelRatio);
-      const keepMod = Math.ceil(tickLabels.length / maxCount);
-
-      return tickLabels.map((label, i) => (i % keepMod === 0 ? label : null));
-    },
+    incrs: isOrdinalX ? [1] : useLogScale ? undefined : histogramBucketSizes,
+    splits: useLogScale || isOrdinalX ? undefined : xSplits,
+    values: isOrdinalX
+      ? (u: uPlot, splits: any[]) => splits
+      : (u: uPlot, splits: any[]) => {
```

Comment:
@leeoniya can we remove the explicit typing of `u` and `splits` here? they get inferred correctly as `uPlot` and `number[]` respectively 🥳

Reply:
if TS is happy, then :+1: